test/tests/functional/pbs_cray_suspend_resume.py

```diff
@@ -93,3 +97,365 @@ def test_exclusive_job_not_suspended(self):
+
+        self.server.delete(j4id)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j3id)
+
+    def test_job_accounted_limit_usage(self):
+        """
+        Test that a job is correctly accounted limit usage
+        """
+        try:
+            self.server.status(QUEUE, id='expressq', logerr=False)
+            self.server.manager(MGR_CMD_DELETE, QUEUE, None, 'expressq')
+        except (PbsStatusError, PbsManagerError):
+            pass
+
+        a = {'queue_type': 'execution',
+             'priority': 150,
+             'enabled': 'True',
+             'started': 'True'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, 'expressq')
+
+        a = {'max_run_res_soft.ncpus': '[p:PBS_GENERIC=1]',
+             'eligible_time_enable': 1}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        j1 = Job(TEST_USER, {ATTR_l + '.select': '1:ncpus=1',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'A'})
+        j1id = self.server.submit(j1)
+
+        # Ensure j2 is started after j1 due to preempt_sort:
+        # min_time_since_start
+        time.sleep(1)
+        d = self.server.expect(JOB, {ATTR_state: 'R'}, id=j1id)
+        self.assertTrue(d)
+
+        j2 = Job(TEST_USER, {ATTR_l + '.select': '1:ncpus=1',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'A'})
+        j2id = self.server.submit(j2)
+        d = self.server.expect(JOB, {ATTR_state: 'R'}, id=j2id)
+        self.assertTrue(d)
+
+        j3 = Job(TEST_USER, {ATTR_l + '.select': '1:ncpus=1',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'B',
+                             ATTR_queue: 'expressq'})
+        j3id = self.server.submit(j3)
+        d = self.server.expect(JOB, {ATTR_state: 'R'}, id=j3id)
+        self.assertTrue(d)
+        d = self.server.expect(JOB, {ATTR_state: 'S'}, id=j2id)
+        self.assertTrue(d)
+
+        j4 = Job(TEST_USER, {ATTR_l + '.select': '1:ncpus=1',
+                             ATTR_l + '.walltime': '3600',
+                             ATTR_project: 'B'})
+        j4id = self.server.submit(j4)
+        d = self.server.expect(JOB, {ATTR_state: 'Q'}, id=j4id)
+
+    def test_preempted_topjob_calendared(self):
+        """
+        That even if topjob_ineligible is set for
+        a preempted job and sched_preempt_enforce_resumption
+        is set true , the preempted job will be calendared
+        """
+
+        self.server.manager(MGR_CMD_SET, SCHED,
+                            {'sched_preempt_enforce_resumption': 'true'})
+        self.server.manager(MGR_CMD_SET, SERVER, {'backfill_depth': '2'})
+
+        nodes = self.server.status(NODE)
+        for n in nodes:
+            if 'resources_available.vntype' in n:
+                if n['resources_available.vntype'] == 'cray_compute':
+                    ncpus = n['resources_available.ncpus']
```

Comment:
Looks like you forgot to remove this ncpus code.  Remember to adjust the job request too.